File: Android/App/app/src/main/java/com/efalco777/crud_products/domain/product/model/ProductDetails.kt
package com.efalco777.crud_products.domain.product.model

data class ProductDetails(
    val id: Int,
    val title: String,
    val description: String,
    val category: String,
    val price: Double,
    ) {
    companion object {
        // Factory function for creating an empty or default instance
        fun empty() = ProductDetails(
            id = 0,
            title = "",
            description = "",
            category = "",
            price = 0.0
        )
    }
}

File: Flutter/App/lib/domain/product/model/product_details.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product_details.freezed.dart';

@freezed
class ProductDetails with _$ProductDetails {
  const factory ProductDetails({
    required int id,
    required String title,
    required String description,
    required double price,
    required String category,
    required List<String> images,
  }) = _ProductDetails;

  static const empty = ProductDetails(
    id: 0,
    title: '',
    description: '',
    price: 0,
    category: '',
    images: [],
  );
}
